EventCart - Move 'Pending in cart' status to extension

colemanw opened the pull request:
Overview
----------------------------------------
Moves a participant status type out of core and into the extension where it belongs.

Comments
-------------
I was really trying to get rid of this hack:

https://github.com/civicrm/civicrm-core/blob/19a5745c09f693c66ffb0047f55f61e7e9072e0b/CRM/Event/BAO/Participant.php#L1760-L1766

But I got stuck on the fact that the extension cannot be disabled in the UI, which means the managed entity won't auto enable/disable the record, which means we still need the hack.

@eileenmcnaughton I think the next step for event cart is to make the extension visible in the UI, and remove the `enable_cart` setting, per 

https://github.com/civicrm/civicrm-core/blob/85c643be0ac3341dd3a05ae5740f88676adbf21a/ext/eventcart/README.md#L7-L11

colemanw committed:
EventCart - Move 'Pending in cart' status to extension

eileenmcnaughton commented:
Oh - I thought we had disabled it for everyone without the setting set earlier on - but can't find that now

colemanw commented:
Yea @eileenmcnaughton see https://github.com/civicrm/civicrm-core/pull/30525#issuecomment-2317993234

colemanw committed:
Api4 - Ensure 'weight' column is not required for SortableEntities

Api4 auto-manages the weight of SortableEntities, so the field is never required.

colemanw commented:
@eileenmcnaughton it turns out the test failure was a real bug (and real annoying to track down).
The `ParticipantStatusType.weight` column is required in the schema and has no default, so the managedEntity was crashing on a required field error.
Fixed now.
